--- a/src/src/core_helpers.c
+++ b/src/src/core_helpers.c
@@ -88,13 +88,10 @@
 #endif /* BSD specific code. */
 
 	mem = mmap(NULL, size, (PROT_READ | PROT_WRITE),
-	    (flags |
+	    (flags
 #ifdef BSD /* BSD specific code. */
 	    MAP_ALIGNED_SUPER
 #endif /* BSD specific code. */
-#ifdef __linux__ /* Linux specific code. */
-	    MAP_HUGETLB
-#endif /* Linux specific code. */
 	    ), fd, 0);
 	if (MAP_FAILED == mem) { /* Retry without super/huge pages */
 		mem = mmap(NULL, size, (PROT_READ | PROT_WRITE), flags, fd, 0);
--- a/src/src/core_thrp.c
+++ b/src/src/core_thrp.c
@@ -666,7 +666,7 @@
 	if ((uintptr_t)-1 == thrpt->io_fd)
 		return (errno);
 	/* Init threads message exchange. */
-	if (-1 == pipe2(thrpt->queue_fd, O_NONBLOCK))
+	if (-1 == pipe(thrpt->queue_fd))
 		return (errno);
 	thrpt->queue_udata.cb_func = NULL;
 	thrpt->queue_udata.ident = thrpt->queue_fd[0];
