--- a/Makefile
+++ b/Makefile
@@ -6,8 +6,8 @@
 #  Copyright (c) 2005-2014 Jason Bittel <jason.bittel@gmail.com>
 #
 
-CC		= gcc
-CCFLAGS  	= -Wall -O3 -funroll-loops -I/usr/include/pcap -I/usr/local/include/pcap
+CC		?= gcc
+CCFLAGS  	= $(CFLAGS) -Wall -O3 -funroll-loops
 DEBUGFLAGS	= -Wall -g -DDEBUG -I/usr/include/pcap -I/usr/local/include/pcap
 LIBS		= -lpcap -lm -pthread
 PROG		= httpry
@@ -18,7 +18,7 @@
 all: $(PROG)
 
 $(PROG): $(FILES)
-	$(CC) $(CCFLAGS) -o $(PROG) $(FILES) $(LIBS)
+	$(CC) $(CCFLAGS) $(LDFLAGS) -o $(PROG) $(FILES) $(LIBS)
 
 debug: $(FILES)
 	@echo "--------------------------------------------------"
